### DISCUSSION TITLE
Copy stream link from current tab's options

### DISCUSSION THREAD

**Discussion by HiroLeWeeb** [Chatterino/chatterino2#3761] [Ideas] [answered]
I, as a small streamer, use my in-game chat to promote my stream. But I find it a bit "too long" to alt-tab to my browser just to start typing "twitch" to get to copy my stream link (also thinking about me and other streamers with a low-mid tier PC that could get a bit laggy).

Pros : 
- Helps promote your channel (or the channel you're moderating)
- Way faster and memory-efficient way to get your stream link without typing it by hand or alt-tabbing to type it in a browser

Cons : 
- It's basically "open stream in browser" and copying the link, but without opening it.
- ???

It's a pretty small feature, but could save some time for most people that need their stream link on hand.

**Comment by zneix** [comment]
This sounds like a very niche scenario to support.
As of right now, there's a quick way to copy current channel's login name (Ctrl+R brings up channel switcher dialog current channel's name already selected, Ctrl+C to copy and Esc to close the dialog). Then you'd just need to type "twitch.tv/" + Ctrl+V.

I know that what I've proposed isn't a full solution but honestly I don't see a reason to add something so specific when you can just do this yourself - if we were to add every single small request like this then there'd way more unnecessary clutter in options / dialog menu settings (which is already quite messy) for things that you can pretty much do yourself.

P.S. I know you're most likely willing to avoid opening a new tab in browser by middle-clicking split header. (See [chatterino wiki](https://wiki.chatterino.com/Glossary/#anatomy-of-a-chatterino-window) for reference about what I'm talking about).

**Comment by KararTY** [comment]
If this is important to you and you need something right now, I'd recommend using some custom hotkey software. Would allow you to customize things even further.

Also, if your keyboard software supports it, you could just setup a hotkey that types the stream link for you otherwise there's also AHK. gl 👍

**Comment by zneix** [comment]
After another thought I realized I've wanted to implement a helper in-built chatterino command which would copy all it's arguments to clipboard. And think that this could potentially solve your usecase. 
Example of what I have in mind (this does not exist yet, just an idea): 
![slashcopy command](https://cdn.zneix.eu/5zlpSZA.png)

then, by typing "/copystream" in your own chat, it would copy `https://twitch.tv/LOGIN` to clipboard.

**Comment by HiroLeWeeb** [comment]
> This sounds like a very niche scenario to support.

That's why I hesitated bringing this idea.

> Also, if your keyboard software supports it, you could just setup a hotkey that types the stream link for you otherwise there's also AHK. gl 👍

Didn't even think of AHK even though I used it 5 minutes ago. 

> After another thought I realized I've wanted to implement a helper in-built chatterino command which would copy all it's arguments to clipboard. And think that this could potentially solve your usecase. Example of what I have in mind (this does not exist yet, just an idea): ![slashcopy command](https://camo.githubusercontent.com/bc90d029f4d57b8c50a38d1a6ad02ac00f80a51ed96ee939c91c7662bacdc871/68747470733a2f2f63646e2e7a6e6569782e65752f357a6c70535a412e706e67)
> 
> then, by typing "/copystream" in your own chat, it would copy `https://twitch.tv/LOGIN` to clipboard.

That would be a really neat implementation

**Reply by zneix** [reply]
Created GH-3763 to implement my idea, keep an eye on that PR - if it gets merged you can download a new [nightly build](https://wiki.chatterino.com/Help/#what-is-nightly-and-how-to-use-install-it) which should have that feature included.

**Comment by Felanbird** [comment, marked as answer]
Resolved in #3763 - exactly as zneix mentioned, just make a command that uses `/copy https://twitch.tv/{channel.name}`
